uamqp/aio/_connection_async.py

```diff
@@ -450,12 +450,9 @@ async def listen(self, wait=False, batch=1, **kwargs):
                     description="Connection was already closed."
                 )
                 return
-            try:
-                tasks = [asyncio.ensure_future(self._listen_one_frame(**kwargs)) for _ in range(batch)]
-                await asyncio.gather(*tasks)
-            except ValueError:
-                for task in tasks:
-                    task.cancel()
+            for _ in range(batch):
+                if await asyncio.ensure_future(self._listen_one_frame(**kwargs)):
```

Comment:
What's the benefit in using `ensure_future` if we're going to directly `await` the coroutine?

Reply:
in my experiment, ensure_future actually gives better performance than await..
let me add a todo to further test the diff here.